.github/workflows/repo-rulesets.yml

```diff
@@ -0,0 +1,140 @@
+            -H "Authorization: token $ACCESS_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+          $API_URL
+          )
+          team_id=$(echo "$response" | jq -r '.id')
+
+          echo "TEAM_ID=$team_id" >> $GITHUB_ENV
+
+      - name: Update Rulesets
+        env:
+          ACCESS_TOKEN: ${{ secrets.BLUE_ENERGY_TOKEN }}
+          FEATUER_CONFIG_PATH: ".github/repo-feature-ruleset.json"
+          AZURE_CONFIG_PATH: ".github/repo-azure-ruleset.json"
+          ORG_NAME: "EDBlueEnergy"
+          TEAM_ID: ${{env.TEAM_ID}}
+        run: |
+
+          # Get ruleset json
+          FEATURE_RULESET=$(cat $FEATUER_CONFIG_PATH)
+          FEATURE_RULESET_NAME=$(echo "$FEATURE_RULESET" | jq -r '.name')
+
+          AZURE_RULESET=$(cat $AZURE_CONFIG_PATH)
+          AZURE_RULESET_NAME=$(echo "$AZURE_RULESET" | jq -r '.name')
+
+          # Add bypass team
+          UPDATED_FEATURE_RULESET=$(echo "$FEATURE_RULESET" | jq --argjson team_id "$TEAM_ID" '. + {"bypass_actors":[{"actor_id":$team_id,"actor_type":"Team","bypass_mode":"always"}]}')
+          UPDATED_AZURE_RULESET=$(echo "$AZURE_RULESET" | jq --argjson team_id "$TEAM_ID" '. + {"bypass_actors":[{"actor_id":$team_id,"actor_type":"Team","bypass_mode":"always"}]}')
+
+          for repo_json in $(jq -c '.[]' repos.json); do
+            REPO_NAME=$(echo "$repo_json" | jq -r '.name')
+
+            if [ "$REPO_NAME" == ".github" ]; then
+              echo "Skipping .github repo..."
+              continue
+            fi
+
+            # Check create or update ruleset
+            REPO_RULESET_LIST=$(curl -L \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token $ACCESS_TOKEN" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "https://api.github.com/repos/$ORG_NAME/$REPO_NAME/rulesets"
+            )
+
+            FEATURE_RULESET_ID=$(echo "$REPO_RULESET_LIST" | jq -r --arg ruleset_name "$FEATURE_RULESET_NAME" '.[] | select(.name == $ruleset_name) | .id')
+            AZURE_RULESET_ID=$(echo "$REPO_RULESET_LIST" | jq -r --arg ruleset_name "$AZURE_RULESET_NAME" '.[] | select(.name == $ruleset_name) | .id')
+
+            # - APPLY Feature ruleset
+            if [ -z "$FEATURE_RULESET_ID" ]; then
+              # API to CREATE ruleset
+              REPO_URL="https://api.github.com/repos/$ORG_NAME/$REPO_NAME/rulesets"
+              HTTP_METHOD="POST"
+            else
+              # API to UPDATE existing ruleset
+              REPO_URL="https://api.github.com/repos/$ORG_NAME/$REPO_NAME/rulesets/$FEATURE_RULESET_ID"
+              HTTP_METHOD="PUT"
+            fi
+
+            curl -L \
+              -X $HTTP_METHOD \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token $ACCESS_TOKEN" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              -d "$UPDATED_FEATURE_RULESET" \
+              "$REPO_URL"
+
+            # - APPLY Azure ruleset
```

Comment:
Should we protect the default branch as well? e.g. master, main...

Reply:
Automated default branch path added https://github.com/EDBlueEnergy/.github/pull/2/commits/be94ec5b8ca3156fbbbd7887a04aeca974276cb6